### PR TITLE
API Updates

### LINKED ISSUE #1575: API version 2022-08-01 removed Checkout Session Create's line_items[amount] but still available in Typescript API?
### Describe the bug

Hi, I'm upgrading from API `2020-08-27` to `2022-08-01`, and have upgraded my stripe-node to the latest version (`v10.13.0`) specifying `apiVersion: '2022-08-01'`.

The [Upgrade Guide][1] mentions:

> The following parameters have been removed from create Checkout Session: line_items[amount],...

Yet in the Stripe Node Library, those types were not removed from [`Stripe.Checkout.SessionCreateParams`][2]:

https://github.com/stripe/stripe-node/blob/db94c299ce2fa1eeaad8123830ee3108153d5a6c/types/2022-08-01/Checkout/Sessions.d.ts#L1726-L1729

### To Reproduce

Upgrade API to 2022-08-01 and in typescript use:

```
import Stripe from 'stripe';
const stripe = new Stripe(key, {
  apiVersion: '2022-08-01'
});
const params: Stripe.Checkout.SessionCreateParams = {
    cancel_url: "",
    success_url: "",
    billing_address_collection: 'auto', //'required',
    customer: "",
    locale: 'auto',
    mode: 'payment',
    line_items: [
      {
        amount: 100,
        currency: 'USD',
        name: "",
        quantity: 1,
        description: "",
        images: ""
      },
    ],
}
```

### Expected behavior

If the 2022-08-01 API removed Create Checkout Session's `line_items` fields, then it should have been removed from the typescript definitions?

### Code snippets

The API returns:

> You cannot use `line_items.amount`, `line_items.currency`, `line_items.name`, `line_items.description`, or `line_items.images` in this API version. Please use `line_items.price` or `line_items.price_data`. Please see https://stripe.com/docs/payments/checkout/migrating-prices for more information.

### OS

macOS

### Node version

Node v16.14.2

### Library version

stripe-node v10.13.0

### API version

2022-08-01

### Additional context

[1]: https://stripe.com/docs/upgrades#2022-08-01
[2]: https://github.com/stripe/stripe-node/blob/db94c299ce2fa1eeaad8123830ee3108153d5a6c/types/2022-08-01/Checkout/Sessions.d.ts#L1726-L1729

### PR DESCRIPTION
**Pull request by richardm-stripe**
Codegen for openapi v202.
cc @stripe/api-libraries

## Changelog
* Add support for new values `invalid_representative_country` and `verification_failed_residential_address` on enums `Account.future_requirements.errors[].code`, `Account.requirements.errors[].code`, `Capability.future_requirements.errors[].code`, `Capability.requirements.errors[].code`, `Person.future_requirements.errors[].code`, and `Person.requirements.errors[].code`
* Add support for `request_log_url` on `Invoice.last_finalization_error`, `PaymentIntent.last_payment_error`, `SetupAttempt.setup_error`, `SetupIntent.last_setup_error`, `StripeErrorResponse.error`, and `StripeError`
* Add support for `network_data` on `Issuing.Authorization`
* Remove `currency`, `description`, `images`, and `name` from `Checkout.SessionCreateParams` (fixes #1575)

### CODE REVIEW

File: types/2022-08-01/index.d.ts [hunk 1]
```diff
@@ -134,7 +134,7 @@ declare module 'stripe' {
   export class Stripe {
     static Stripe: typeof Stripe;
 
-    constructor(apiKey: string, config: Stripe.StripeConfig);
+    constructor(apiKey: string, config?: Stripe.StripeConfig);
```
Comment (yejia-stripe) [inline types/2022-08-01/index.d.ts:137]: missing from changelog

**Review by yejia-stripe — APPROVED** [review]
lgtm!

File: types/2022-08-01/index.d.ts (hunk 1 above)
Comment (richardm-stripe) [reply types/2022-08-01/index.d.ts:137]: ohh hmm what's this about?

Comment (richardm-stripe) [reply types/2022-08-01/index.d.ts]: I'm going to manually revert this, we want to discuss this a little more before we proceed.

**Comment by ahoisl** [comment]
The commit message and consequently the release notes are a bit misleading with regards to the removed members. These members are not removed from `Checkout.SessionCreateParams`, but from `Chechkout.SessionCreateParams.Discount`.

Almost had a heart attack when reading the release notes 😉

**Pull request merged**